language/move-lang/tests/functional/typing/main_arguments_invalid.move

```diff
@@ -0,0 +1,35 @@
+module M {
+    resource struct R {}
+    struct S {}
+    struct Cup<T> {}
+
+    public fun eat(r: R) {
+        R{} = r
+    }
+}
+
+//! new-transaction
+
+use {{default}}::M;
+
+fun main(
+    no1: M::R,
+    no2: M::S,
+    no3: M::Cup<u64>,
+    no4: M::Cup<M::S>,
+) {
+
+}
+
+// check: error
```

Comment:
Ah, this is how I could have written a test for https://github.com/libra/libra/pull/2716.... I'm not sure if it's important to add that, but at least I see how to do it.